generators/entity/templates/src/main/webapp/app/entities/pagination-template.ejs

```diff
@@ -1,23 +1,30 @@
-        $scope.<%= entityInstancePlural %> = [];
-        $scope.predicate = 'id';
-        $scope.reverse = true;
-        $scope.page = 1;
+        $scope.predicate = pagingParams.predicate;
+        $scope.reverse = pagingParams.ascending;
         <%_ if (fieldsContainBlob) { _%>
         $scope.abbreviate = DataUtils.abbreviate;
         $scope.byteSize = DataUtils.byteSize;
         <%_ } _%>
         $scope.loadAll = function() {
-            <%= entityClass %>.query({page: $scope.page - 1, size: paginationConstants.itemsPerPage, sort: [$scope.predicate + ',' + ($scope.reverse ? 'asc' : 'desc'), 'id']}, function(result, headers) {
+            var onSuccess = function (data, headers) {
                 $scope.links = ParseLinks.parse(headers('link'));
                 $scope.totalItems = headers('X-Total-Count');
-                $scope.<%= entityInstancePlural %> = result;
-            });
+                $scope.<%= entityInstancePlural %> = data;
+                $scope.page = pagingParams.page;
+            };
+            var onError = function (error) {
+                AlertService.error(error.data.message);
```

Comment:
Ok lets merge this and take acre of i18n

Check how validation messages are translated to get an idea